hw/ip/dma/doc/programmers_guide.md

```diff
@@ -0,0 +1,93 @@
+A potential use case for chunked memory-to-memory transfers is memory initialization, where a small chunk of data (e.g., a block of zeros) can be repeatedly transferred to a larger memory region.
+
+## Hardware Handshaking Mode
+
+The DMA supports a hardware handshaking mode that enables seamless data transfers between IO peripherals and memory.
+This mode leverages the chunked data transfer mechanism and interrupts from the IO peripheral.
+
+In this mode, the IO peripheral fills its internal transfer FIFO and then signals the DMA by raising an interrupt.
+Upon receiving this interrupt, the DMA reads the data from the peripheral's FIFO and writes it to the destination memory.
+This process continues in a loop: the DMA transfers a chunk of data, waits for the IO peripheral to fill its FIFO and issue another interrupt, and then transfers the next chunk.
+This loop repeats until the total number of bytes specified by [`TOTAL_DATA_SIZE`](registers.md#total_data_size) has been transferred.
+
+To enable hardware handshaking, software must set the `hardware_handshake_enable` bit in the [`CONTROL`](#control) register and configure the appropriate addressing mode to access the IO peripheral.
+
+The DMA's chunked transfer mechanism in hardware handshaking mode supports two address update modes for the peripheral:
+
+  * **Continuous Access:** The DMA continuously accesses the same address for each chunk.
+  * **Wrap-around:** After transferring a chunk, the DMA's peripheral address wraps around to the starting address configured for the transfer.
+
+For interrupt handling in hardware handshaking mode, software needs to enable the corresponding interrupt using the [`HANDSHAKE_INTR_ENABLE`](#handshake_intr_enable) register.
+
+The DMA also provides a mechanism to acknowledge the interrupt received from the IO peripheral by performing a configurable write operation.
+If interrupt acknowledgement is required, software must enable it in the [`CLEAR_INTR_SRC`](#clear_intr_src) register.
+The address space for this write operation is configured using the [`CLEAR_INTR_BUS`](#clear_intr_bus) register.
+The specific address and data value to be written for acknowledgement are defined by the [`INTR_SRC_ADDR_0-10`](#intr_src_addr) and [`INTR_SRC_WR_VAL_0-10`](#intr_src_wr_val) registers, respectively.
+
+## Inline Hashing
+
+The DMA incorporates an inline hashing capability for SHA-2 algorithms (SHA-256, SHA-384, and SHA-512).
+This allows the DMA to compute the hash digest of the transferred data concurrently with the memory transfer.
+
+To enable inline hashing, software must set the desired SHA-2 algorithm as the opcode in the `opcode` field of the [`CONTROL`](registers.md#control) register.
+This will instruct the DMA to perform the data copy operation along with the hash computation.
+
+When initiating a transfer with inline hashing, the `initial_transfer` bit in the [`CONTROL`](registers.md#control) register must be asserted.
```

Comment:
Whilst this is true, the `initial_transfer` bit must be set at the start of _any DMA transfer_ to indicate that it's the first chunk of the transfer; I think it could be worth clarifying that by describing it further up, perhaps in 'Start the Transfer' or 'Chunked Data Transfers.'